src/library/services/data-provider.service.ts

```diff
@@ -1,8 +1,12 @@
 import {Book} from "../model/book";
+import {SearchCriteria} from "./local-storage-data-provider.service";
```

Comment:
SearchCriteria must be with DataProvider as part of API, not in implementation in local-storage-data-provider.service